src/LuaTransformer.ts

```diff
@@ -104,21 +104,53 @@ export class LuaTransformer {
 
     protected currentSourceFile!: ts.SourceFile;
     protected isModule!: boolean;
+    protected isWithinBundle!: boolean;
     protected resolver!: EmitResolver;
 
     /** @internal */
-    public transform(sourceFile: ts.SourceFile): [tstl.Block, Set<LuaLibFeature>] {
+    public transform(node: ts.Bundle | ts.SourceFile): [tstl.Block, Set<LuaLibFeature>] {
         this.setupState();
-        this.currentSourceFile = sourceFile;
-        this.isModule = tsHelper.isFileModule(sourceFile);
+        if (ts.isSourceFile(node)) {
+            this.currentSourceFile = node;
+            this.isModule = tsHelper.isFileModule(node);
 
-        // Use `getParseTreeNode` to get original SourceFile node, before it was substituted by custom transformers.
-        // It's required because otherwise `getEmitResolver` won't use cached diagnostics, produced in `emitWorker`
-        // and would try to re-analyze the file, which would fail because of replaced nodes.
-        const originalSourceFile = ts.getParseTreeNode(sourceFile, ts.isSourceFile) || sourceFile;
-        this.resolver = this.checker.getEmitResolver(originalSourceFile);
+            // Use `getParseTreeNode` to get original SourceFile node, before it was substituted by custom transformers.
+            // It's required because otherwise `getEmitResolver` won't use cached diagnostics, produced in `emitWorker`
+            // and would try to re-analyze the file, which would fail because of replaced nodes.
+            const originalSourceFile = ts.getParseTreeNode(node, ts.isSourceFile) || node;
+            this.resolver = this.checker.getEmitResolver(originalSourceFile);
 
-        return [this.transformSourceFile(sourceFile), this.luaLibFeatureSet];
+            return [this.transformSourceFile(node), this.luaLibFeatureSet];
+        } else {
+            return [this.transformBundle(node), this.luaLibFeatureSet];
+        }
+    }
+
+    public transformBundle(bundle: ts.Bundle): tstl.Block {
+        this.isWithinBundle = true;
+        const combinedStatements = bundle.sourceFiles.reduce(
+            (statements: tstl.Statement[], sourceFile: ts.SourceFile) => {
+                this.currentSourceFile = sourceFile;
+                this.isModule = tsHelper.isFileModule(sourceFile);
+                const originalSourceFile = ts.getParseTreeNode(sourceFile, ts.isSourceFile) || sourceFile;
+                this.resolver = this.checker.getEmitResolver(originalSourceFile);
+                const transformResult = this.transformSourceFile(sourceFile);
+                return [...statements, ...transformResult.statements];
+            },
+            []
+        );
+
+        if (this.options.luaEntry) {
+            const requireCalls = this.options.luaEntry.map(entry => {
+                const formattedEntryName = tsHelper.formatPathToLuaPath(entry.replace(/.ts$/, ""));
```

Comment:
That looks very loose. I think we should decide either to do a full path resolution (i.e. `./src/file.ts` -> `file`, make it relative to tsconfig file location) or don't transform it at all. I personally lean towards the second option now.